### DISCUSSION TITLE
Experimental "redirects" feature not redirecting to expected url (missing "query" part)

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/next.js#12391] [Help]
Using the following in`next.config.js`

```js
experimental: {
    redirects() {
      return [
        {
          source: '/:lang((?!fr|en))/:path*',
          destination: '/api/autoRedirectToLocalisedPage?path=:path*',
          // Permanent uses 308, otherwise uses 307 - See https://github.com/zeit/next.js/issues/9081
          permanent: process.env.APP_STAGE === 'production', // Do not use permanent redirect locally to avoid browser caching when working on it
        },
      ];
    },
  },
```

1. When I go to `http://localhost:8888/`, I get redirected to `http://localhost:8888/api/autoRedirectToLocalisedPage`, but I'd expect to be redirected to `http://localhost:8888/api/autoRedirectToLocalisedPage?path=`. 

1. When I go to `http://localhost:8888/test/some-page`, I don't get redirected at all and it displays a 404 page. 

Am I doing something wrong? I want that to redirect if the url doesn't start with `/fr` or `/en`.

**Comment by TommySorensen** [comment]
This is apparently out of scope of RFC
https://github.com/zeit/next.js/issues/9081#issuecomment-629865019
